model/githubapp/github_app_installation.go

```diff
@@ -243,77 +243,20 @@ func getInstallationIDFromGitHub(ctx context.Context, authFields *GithubAppAuth,
 	return installation.GetID(), nil
 }
 
-// cachedInstallationToken represents a GitHub installation token that's
-// cached in memory.
-type cachedInstallationToken struct {
-	installationToken string
-	expiresAt         time.Time
-}
-
-func (c *cachedInstallationToken) isExpired(lifetime time.Duration) bool {
-	return time.Until(c.expiresAt) < lifetime
-}
-
-// installationTokenCache is a concurrency-safe cache that maps a unique key
-// (composed of the installation ID and the token's permissions) to the cached
-// GitHub installation token.
-type installationTokenCache struct {
-	cache map[string]cachedInstallationToken
-	mu    sync.RWMutex
-}
-
 // ghInstallationTokenCache is the in-memory instance of the cache for GitHub
 // installation tokens.
-var ghInstallationTokenCache = installationTokenCache{
-	cache: make(map[string]cachedInstallationToken),
-	mu:    sync.RWMutex{},
-}
-
-// get gets an installation token from the cache by its installation ID and permissions. It will
-// not return a token if the token will expire before the requested lifetime.
-func (c *installationTokenCache) get(installationID int64, permissions *github.InstallationPermissions, lifetime time.Duration) string {
-	c.mu.RLock()
-	defer c.mu.RUnlock()
-	id, err := createCacheID(installationID, permissions)
-	if err != nil {
-		return ""
-	}
-	cachedToken, ok := c.cache[id]
-	if !ok {
-		return ""
-	}
-	if cachedToken.isExpired(lifetime) {
-		return ""
-	}
-
-	return cachedToken.installationToken
-}
+var ghInstallationTokenCache2 = cache.New[string]()
```

Comment:
Is there a reason this is named `Cache2`?

Reply:
Because the OG cache was still around when I was replacing it